components/misc/stringops.hpp

```diff
@@ -56,6 +57,23 @@ class StringUtils
         };
     }
 
+    static uint32_t toUpper(uint32_t ch)
+    {
+        // Russian alphabet
+        if (ch >= 0x0430 && ch < 0x0450)
+            ch -= 0x20;
+
+        // Cyrillic YO character
+        if (ch == 0x0451)
+            ch -= 0x50;
+
+        // Latin alphabet
+        if (ch >= 0x61 && ch < 0x80)
+            ch -= 0x20;
+
+        return ch;
+    }
```

Comment:
I think I would have put this function somewhere locally in the journal code, because it's not complete enough to be generally useful and because the inconsistency with StringUtils::toLower() (which deals with ASCII) is confusing.